### PR TITLE
Change time from duckdb_time to duckdb_time_struct in duckdb_time_tz_struct

### PR DESCRIPTION
**Pull request by Giorgi**
I tried to implement https://github.com/duckdb/duckdb/discussions/10917

### CODE REVIEW

**Comment by Giorgi** [comment]
@Mytherin @Mause Can you review?

File: src/main/capi/datetime-c.cpp [hunk 1]
```diff
@@ -45,9 +45,19 @@ duckdb_time_struct duckdb_from_time(duckdb_time time) {
...
 
 duckdb_time_tz_struct duckdb_from_time_tz(duckdb_time_tz input) {
-	duckdb::dtime_tz_t time(input.bits);
 	duckdb_time_tz_struct result;
-	result.time.micros = time.time().micros;
+	duckdb_time_struct time_struct;
+	int32_t hour, minute, second, micros;
+
+	duckdb::dtime_tz_t time(input.bits);
+	Time::Convert(time.time(), hour, minute, second, micros);
```
Comment (Mytherin) [inline src/main/capi/datetime-c.cpp:53]: Can this use `duckdb_from_time` directly instead?

**Review by Mytherin — COMMENTED** [review]
Thanks! LGTM - one comment

File: src/main/capi/datetime-c.cpp (hunk 1 above)
Comment (Giorgi) [reply src/main/capi/datetime-c.cpp]: Done.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**